src/server/components/dashboard/listsItems/item/update/actionHandlers/helpers.ts

```diff
@@ -17,3 +21,34 @@ export async function sendPublishedEmail(listItem: ListItemWithAddressCountry) {
     searchLink
   );
 }
+
+export async function sendUnpublishEmail(listId: number) {
+  const list = (await prisma.list.findFirst({
+    where: {
+      id: listId,
+    },
+    include: {
+      country: true,
+    },
+  })) as List;
+
+  const users = list.jsonData.users as string[];
+
+  if (!users) {
+    logger.error(`No users found for list ${listId}`);
+    return;
+  }
+
+  logger.info(`Sending unpublish email to ${users.length} users`);
+
+  if (users) {
+    const tasks = users.map(async (user) => {
+      await sendManualUnpublishedEmail({
+        emailAddress: user,
+        serviceType: lowerCase(startCase(list.type)),
+        country: list.country?.name as string,
```

Comment:
do all 3 of the emails require the same personalisations? just different template? If so, then it's worth abstracting this. `sendManualActionNotificationToPost(listId, action)`?

Reply:
I feel like it might make more sense to abstract the functions in govulk-notify.ts instead. Maybe I can do it in both places 😁

Reply:
Well actually, maybe not the govuk-notify stuff. There are too many unique strings between the functions to make it worth it 🤷